src/main/java/org/apache/sling/auth/oauth_client/impl/OidcAuthenticationHandler.java

```diff
@@ -159,11 +168,13 @@ public OidcAuthenticationHandler(@Reference(policyOption = ReferencePolicyOption
         this.stateManager = stateManager;
         this.idp = config.idp();
         this.callbackUri = config.callbackUri();
-        this.defaultRedirect = config.defaultRedirect();
         this.loginCookieManager = loginCookieManager;
         this.defaultConnectionName = config.defaultConnectionName();
         this.userInfoProcessor = userInfoProcessor;
         this.userInfoEnabled = config.userInfoEnabled();
+        this.checkNonce = config.checkNonce();
```

Comment:
What is this in reference to? I browsed https://datatracker.ietf.org/doc/html/rfc6749 but don't see anything special (except for a non-normative example in https://datatracker.ietf.org/doc/html/rfc6749#section-7.1 ).

Reply:
Oh, this is the OIDC handler, right. Quoting from the spec

>  If present in the ID Token, Clients MUST verify that the nonce Claim Value is equal to the value of the nonce parameter sent in the Authentication Request.

What is the reason for making this configurable?

Reply:
the nonce parameter is not mandatory in the request, but if it is present the IdP must send it in the ID Token.
I made it configurable in case some IdP do not support it.

Reply:
So you are concerned about sending a nonce in the authentication request to an authorization server that does not support nonces?

Reply:
Yes. If the server would not support it, we would not validate the token.

Reply:
I realise I'm being pedantic, but I don't like seeing config options that should not be there :-) You don't have to remove if you're really sure that there are going to be non-compliant OIDC server implementations out there we need to support.

Otherwise I would always keep it enabled because it enhances security and disabling it would bring no benefit.

Reply:
I removed that configuration.